src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ExtendedJob2dAdapter.java

```diff
@@ -0,0 +1,53 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+
+public class ExtendedJob2dAdapter extends DefaultJob2dAdapter {
+
+    public enum Line {
+        CONTINUOUS,
+        DOTTED,
+        SPECIAL
+    }
+
+    private Line line = Line.SPECIAL;
+
+    public ExtendedJob2dAdapter() {
+        super();
+    }
+
+    public void setLine(Line type) {
+        this.line = type;
+    }
+
+    private ILine getLine() {
+        switch (this.line) {
+            case DOTTED:
+                return LineFactory.getDottedLine();
+            case SPECIAL:
+                return LineFactory.getSpecialLine();
+            default:
+                return LineFactory.getBasicLine();
+        }
+    }
```

Comment:
OK, but coupling with LineFactory is not necessary (with e.g., constructor injection)